evalml/pipelines/utils.py

```diff
@@ -657,6 +667,11 @@ def _make_new_component_name(name, component_name, idx=None, pipeline_name=None)
             new_component_name = _make_new_component_name(
                 component_pipeline_name, name, name_idx, sub_pipeline_name
             )
+            first_x_component = (
+                pipeline.component_graph.compute_order[0]
+                if pipeline.component_graph.compute_order[0] != "Label Encoder"
```

Comment:
AFAIK this is the only case where a component will come out of `make_pipelines` with a component that alters y as the very first component. I could add more logic here but that would involve instantiating a component and checking if it affects x or y.